drivebrain_app/DriveBrainApp.cpp

```diff
@@ -0,0 +1,239 @@
+        // samples internal state set by the handle recv functions
+
+        auto state_and_validity = state_estimator_->get_latest_state_and_validity();
+        auto out_struct = controller_->step_controller(state_and_validity.first);
+        auto temp_desired_torques = state_and_validity.first.matlab_math_temp_out;
+        // feedback
+        state_estimator_->set_previous_control_output(out_struct);
+        // output
+        // if(state_and_validity.second)
+        // {
+        out_msg->set_prev_mcu_recv_millis(out_struct.mcu_recv_millis);
+
+        if(temp_desired_torques.res_torque_lim_nm.FL < 0) {
+            out_msg->mutable_desired_rpms()->set_fl(0);
+        } else {
+            out_msg->mutable_desired_rpms()->set_fl(out_struct.desired_rpms.FL);
+        }
+
+        if(temp_desired_torques.res_torque_lim_nm.FR < 0) {
+            out_msg->mutable_desired_rpms()->set_fr(0);
+        } else {
+            out_msg->mutable_desired_rpms()->set_fr(out_struct.desired_rpms.FR);
+        }
+
+        if(temp_desired_torques.res_torque_lim_nm.RL < 0) {
+            out_msg->mutable_desired_rpms()->set_rl(0);
+        } else {
+            out_msg->mutable_desired_rpms()->set_rl(out_struct.desired_rpms.RL);
+        }
+
+        if(temp_desired_torques.res_torque_lim_nm.RR < 0) {
+            out_msg->mutable_desired_rpms()->set_rr(0);
+        } else {
+            out_msg->mutable_desired_rpms()->set_rr(out_struct.desired_rpms.RR);
+        }
+
+        out_msg->mutable_torque_limit_nm()->set_fl(::abs(temp_desired_torques.res_torque_lim_nm.FL));
+        out_msg->mutable_torque_limit_nm()->set_fr(::abs(temp_desired_torques.res_torque_lim_nm.FR));
+        out_msg->mutable_torque_limit_nm()->set_rl(::abs(temp_desired_torques.res_torque_lim_nm.RL));
+        out_msg->mutable_torque_limit_nm()->set_rr(::abs(temp_desired_torques.res_torque_lim_nm.RR));
+
+        {
+            std::unique_lock lk(eth_tx_queue_.mtx);
+            eth_tx_queue_.deque.push_back(out_msg);
+            eth_tx_queue_.cv.notify_all();
+        }
+
+        auto end_time = std::chrono::high_resolution_clock::now();
+        auto elapsed = std::chrono::duration_cast<std::chrono::microseconds>(end_time - start_time);
+        
+        if (loop_chrono_time > elapsed) {
+            std::this_thread::sleep_for(loop_chrono_time - elapsed);
+        }
+    }
+}
+
+void DriveBrainApp::startThreads() {
+    db_service_thread_ = std::make_unique<std::thread>([this]() {
+        spdlog::info("started db service thread");
+        try {
+            while (!stop_signal_.load()) {
+                // Run the io_context as long as stop_signal is false
+                db_service_->run_server(); // Run at least one handler, or return immediately if none
+            }
+        } catch (const std::exception& e) {
+            spdlog::error("Error in drivebrain service thread: {}", e.what());
+        }
+    });
```

Comment:
this can also be within the constructor itself. I would also suggest allow configuration of the DriveBrainApp class through constructor arguments to allow for disabling / enabling running of different threads, for instance we could have a constructor argument called `settings` that is a struct with members such as `bool run_service;` where if this is true, this thread is created, otherwise it is not created.

Reply:
also, instead of heap allocating the thread, please use an initializer list for construction of these threads so you dont have to use `std::make_unique` for dynamic allocation of the threads as we do not need them to be.